lib/bap_disasm/bap_disasm_symtab.ml

```diff
@@ -28,6 +28,7 @@ module Fn = Opaque.Make(struct
 type t = {
   addrs : fn Addr.Map.t;
   names : fn String.Map.t;
+  calls : fn Addr.Map.t;
```

Comment:
what is the justification to put this into the Symtab? This adds a completely new responsibility for this data structure, and we need to review each function in its implementation that it preserves its invariants. For example, you at least broke the `remove` function. The mere fact that you didn't touch it means, that it is broken, as what will happen if you will remove a function that is in the calls?

Reply:
Justification is:
we lose information about calls in reconstructor when we take apart a whole program cfg, that is very harmful in case of relocations. And the `Symtab` looks like a good place to store this information. 
And we definitely need to review each function, what I could do incorrect, but `remove` function, as far as I see, removes calls, therefore I did touch it

Reply:
So, you actually did touch it. 
What do you mean by "harmful in case of relocations"?
And yes, I agree, that we lose information when we proceed from the whole program CFG to a set of function CFGs, and since we pass only symtab downstream we need to preserve this information. 

I have a couple more questions:

Does the `calls` data structure relates callsites to the called function?
Is it guaranteed that domain of call is equal to the domain of addrs ? (I.e., that forall x, exists y, calls[x] = addrs[y])